packages/matter.js/src/behavior/system/network/NetworkServer.ts

```diff
@@ -101,21 +99,156 @@ export class NetworkServer extends NetworkBehavior {
+                        }
+                        listen.push({ protocol, address });
+                        this.state.ble = true;
+                    } else {
+                        disabledBle = true;
+                    }
+                    break;
+
+                case "udp":
+                case "udp4":
+                case "udp6":
+                    hasUdp = true;
+                    if (port === undefined) {
+                        port = this.state.port;
+                    }
+                    listen.push({ protocol, address, port });
+                    break;
+
+                default:
+                    throw new ImplementationError(`Unknown listen protocol "${protocol}"`);
+            }
+        }
+
+        if (disabledBle) {
+            logger.warn("Disabling Bluetooth commissioning because BLE support is not installed");
+            this.state.ble = false;
+        } else if (this.state.ble !== false && Ble.enabled) {
+            if (!hasBle) {
+                listen.push({ protocol: "ble" });
+            }
+            this.state.ble = true;
+        }
+
+        if (!hasUdp) {
+            listen.push({ protocol: "udp", port: this.state.port });
+        }
+
+        return listen;
+    }
 }
 
 export namespace NetworkServer {
+    /**
+     * A UDP listening address.
+     */
+    export interface UdpAddress {
+        protocol: "udp" | "udp4" | "udp6";
+
+        /**
+         * The hostname or IP address.  Leave undefined for all addresses, "0.0.0.0" for all IPv4 addresses, and "::"
+         * for all IPv6 addresses.
+         */
+        address?: string;
+
+        /**
+         * The port to listen on.  Defaults to {@link State.port}.
+         */
+        port?: number;
+    }
+
+    /**
+     * A Bluetooth LE listening address,
+     *
+     * TODO - currently only a single BLE transport is supported
+     */
+    export interface BleAddress {
+        protocol: "ble";
+
+        /**
+         * The HCI ID of the bluetooth adapter.
+         *
+         * By default selects the first adapter on the system.
+         *
+         * TODO - currently you cannot specify HCI ID here
+         */
+        address?: string;
+    }
+
+    export type Address = BleAddress | UdpAddress;
+
     export class Internal extends NetworkBehavior.Internal {
         declare runtime: ServerNetworkRuntime;
     }
 
     export class State extends NetworkBehavior.State {
-        listeningAddressIpv4?: string = undefined;
-        listeningAddressIpv6?: string = undefined;
-        ipv4 = true;
+        /**
+         * An array of {@link Address} objects configuring the interfaces the server listens on.
+         *
+         * Configurable also with variable "network.listen".  You may configure a single listener using:
+         *
+         *   * `network.listen.protocol` either "ble", "udp4", "udp6" or "udp" (default is "udp" for dual IPv4/6)
```

Comment:
we need to see where we document all these ... I started as main comment also on Environment  (in this case the NodeEnvironment). So we need to update there.

Reply:
I think in a separate .md file would be ideal, I've been planning to document in the code near where the options are used.  Environment seems to generic to document the vars, many of which will be irrelevant if you're a controller vs. device.

This was one of the things I was going to address in the documentation push.